### PR TITLE
Add --gdal-version command line option to fio

### PR DESCRIPTION
**Pull request by micahcochran**
This command line option gives the GDAL Version that Fiona is using when running `fio` commands.

Console example on Linux:
```
$ fio env --gdalversion
GDAL Version is 1.11.2
```

There could possibly be a use case if have multiple installations of GDAL on a system.  I was using this on Travis to figure out if the correct GDAL version was being used by Fiona.  I'm not sure if this is really useful.

### CODE REVIEW

**Comment by sgillies** [comment]
@micahcochran I think it's useful! Can you align it with https://github.com/mapbox/rasterio/blob/master/rasterio/rio/main.py#L66-L68 ?

**Comment by micahcochran** [comment]
@sgillies   I reverted the original commit.  Then, I copied and pasted the Rasterio code.

Command line now works like rasterio's
```
$ fio --gdal-version
1.11.3
```

Let me know if that is not what you meant by "align".

It looks like the tests might are failing because this has changed expected argument variables for a few of the tests.

File: fiona/fio/main.py [hunk 1]
```diff
@@ -12,20 +12,29 @@
...
 @with_plugins(ep for ep in list(iter_entry_points('fiona.fio_commands')) +
               list(iter_entry_points('fiona.fio_plugins')))
 @click.group()
 @verbose_opt
 @quiet_opt
 @click.version_option(fio_version)
+@click.option(
+    '--gdal-version', is_eager=True, is_flag=True, callback=gdal_version_cb)
 @click.pass_context
 def main_group(ctx, verbose, quiet):
```
Comment (sgillies) [inline fiona/fio/main.py]: @micahcochran: add a `gdal_version` argument to `main_group()`, after `quiet`, and all will be good.

**Comment by micahcochran** [comment]
@sgillies I took a slightly different approach.  This is my attempt to use `@click.version_option` to implement `--gdal-version` and also a `--python-version`.

```
micah@micah-VirtualBox:~/prj/Fiona$ fio --gdal-version
GDAL, version 1.11.3
micah@micah-VirtualBox:~/prj/Fiona$ fio --python-version
Python, version 3.5.1+ (default, Mar 30 2016, 22:46:26) 
[GCC 5.3.1 20160330]
micah@micah-VirtualBox:~/prj/Fiona$ fio --version
fio, version 1.7.0
```

**Comment by sgillies** [comment]
Great!

**Comment by micahcochran** [comment]
Thank you!

**Pull request merged**